File: backend/app/schemas/research.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.schemas.ontology import OntologyResponse


class ResearchTriggerRequest(BaseModel):
    topic: str
    additional_context: str | None = None


class ResearchResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    project_id: uuid.UUID
    ontology_id: uuid.UUID | None  # null while a task is still RUNNING or FAILED
    topic: str | None  # the original user-supplied topic (preserved verbatim)
    report_file_path: str | None
    created_at: datetime
    ontology: OntologyResponse | None = None

File: frontend/src/types/research.ts
import type { OntologyResponse } from './ontology'

export interface ResearchTriggerRequest {
  topic: string
  additional_context?: string | null
}

export interface ResearchResponse {
  id: string
  project_id: string
  ontology_id: string
  report_file_path: string | null
  created_at: string
  ontology: OntologyResponse | null
}
